How to change the S2I base image from the default "fabric8/s2i-java:2.3"

I get the following error when trying to use the "registry.access.redhat.com/redhat-openjdk-18/openjdk18-openshift:latest" builder image. 

[ERROR] Failed to execute goal org.eclipse.jkube:oc-maven-plugin:0.1.1:build (default-cli) on project sample: Failed to execute the build: Unable to build the image using the OpenShift build service: NullPointerException -> [Help 1]

```
<generator>
    <includes>
	<include>spring-boot</include>
   </includes>
    <config>
	<spring-boot>
		<alias>spring-boot</alias>
		<from>registry.access.redhat.com/redhat-openjdk-18/openjdk18-openshift:latest</from>
	</spring-boot>
    </config>
</generator>
```

This works fine for kubernetes strategy. 

Here is the maven Debug output

```
➜  sample git:(master) ✗ mvn oc:build -P openshift -X                                          
Apache Maven 3.6.1 (d66c9c0b3152b2e69ee9bac180bb8fcc8e6af555; 2019-04-04T14:00:29-05:00)
Maven home: /usr/local/Cellar/maven/3.6.1/libexec
Java version: 1.8.0_212, vendor: AdoptOpenJDK, runtime: /Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.15.3", arch: "x86_64", family: "mac"
[DEBUG] Created new class realm maven.api
[DEBUG] Importing foreign packages into class realm maven.api
[DEBUG]   Imported: javax.annotation.* < plexus.core
[DEBUG]   Imported: javax.annotation.security.* < plexus.core
[DEBUG]   Imported: javax.enterprise.inject.* < plexus.core
[DEBUG]   Imported: javax.enterprise.util.* < plexus.core
[DEBUG]   Imported: javax.inject.* < plexus.core
[DEBUG]   Imported: org.apache.maven.* < plexus.core
[DEBUG]   Imported: org.apache.maven.artifact < plexus.core
[DEBUG]   Imported: org.apache.maven.classrealm < plexus.core
[DEBUG]   Imported: org.apache.maven.cli < plexus.core
[DEBUG]   Imported: org.apache.maven.configuration < plexus.core
[DEBUG]   Imported: org.apache.maven.exception < plexus.core
[DEBUG]   Imported: org.apache.maven.execution < plexus.core
[DEBUG]   Imported: org.apache.maven.execution.scope < plexus.core
[DEBUG]   Imported: org.apache.maven.lifecycle < plexus.core
[DEBUG]   Imported: org.apache.maven.model < plexus.core
[DEBUG]   Imported: org.apache.maven.monitor < plexus.core
[DEBUG]   Imported: org.apache.maven.plugin < plexus.core
[DEBUG]   Imported: org.apache.maven.profiles < plexus.core
[DEBUG]   Imported: org.apache.maven.project < plexus.core
[DEBUG]   Imported: org.apache.maven.reporting < plexus.core
[DEBUG]   Imported: org.apache.maven.repository < plexus.core
[DEBUG]   Imported: org.apache.maven.rtinfo < plexus.core
[DEBUG]   Imported: org.apache.maven.settings < plexus.core
[DEBUG]   Imported: org.apache.maven.toolchain < plexus.core
[DEBUG]   Imported: org.apache.maven.usability < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.* < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.authentication < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.authorization < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.events < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.observers < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.proxy < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.repository < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.resource < plexus.core
[DEBUG]   Imported: org.codehaus.classworlds < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.* < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.classworlds < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.component < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.configuration < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.container < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.context < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.lifecycle < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.logging < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.personality < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.util.xml.Xpp3Dom < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.util.xml.pull.XmlPullParser < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.util.xml.pull.XmlPullParserException < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.util.xml.pull.XmlSerializer < plexus.core
[DEBUG]   Imported: org.eclipse.aether.* < plexus.core
[DEBUG]   Imported: org.eclipse.aether.artifact < plexus.core
[DEBUG]   Imported: org.eclipse.aether.collection < plexus.core
[DEBUG]   Imported: org.eclipse.aether.deployment < plexus.core
[DEBUG]   Imported: org.eclipse.aether.graph < plexus.core
[DEBUG]   Imported: org.eclipse.aether.impl < plexus.core
[DEBUG]   Imported: org.eclipse.aether.installation < plexus.core
[DEBUG]   Imported: org.eclipse.aether.internal.impl < plexus.core
[DEBUG]   Imported: org.eclipse.aether.metadata < plexus.core
[DEBUG]   Imported: org.eclipse.aether.repository < plexus.core
[DEBUG]   Imported: org.eclipse.aether.resolution < plexus.core
[DEBUG]   Imported: org.eclipse.aether.spi < plexus.core
[DEBUG]   Imported: org.eclipse.aether.transfer < plexus.core
[DEBUG]   Imported: org.eclipse.aether.version < plexus.core
[DEBUG]   Imported: org.fusesource.jansi.* < plexus.core
[DEBUG]   Imported: org.slf4j.* < plexus.core
[DEBUG]   Imported: org.slf4j.event.* < plexus.core
[DEBUG]   Imported: org.slf4j.helpers.* < plexus.core
[DEBUG]   Imported: org.slf4j.spi.* < plexus.core
[DEBUG] Populating class realm maven.api
[INFO] Error stacktraces are turned on.
[DEBUG] Message scheme: color
[DEBUG] Message styles: debug info warning error success failure strong mojo project
[DEBUG] Reading global settings from /usr/local/Cellar/maven/3.6.1/libexec/conf/settings.xml
[DEBUG] Reading user settings from /Users/Marcelo.Borges@ibm.com/.m2/settings.xml
[DEBUG] Reading global toolchains from /usr/local/Cellar/maven/3.6.1/libexec/conf/toolchains.xml
[DEBUG] Reading user toolchains from /Users/Marcelo.Borges@ibm.com/.m2/toolchains.xml
[DEBUG] Using local repository at /Users/Marcelo.Borges@ibm.com/.m2/repository
[DEBUG] Using manager EnhancedLocalRepositoryManager with priority 10.0 for /Users/Marcelo.Borges@ibm.com/.m2/repository
[INFO] Scanning for projects...
[DEBUG] Extension realms for project com.ibm.gts.aro:sample:jar:0.0.1-SNAPSHOT: (none)
[DEBUG] Looking up lifecycle mappings for packaging jar from ClassRealm[plexus.core, parent: null]
[DEBUG] Extension realms for project org.springframework.boot:spring-boot-starter-parent:pom:2.2.4.RELEASE: (none)
[DEBUG] Looking up lifecycle mappings for packaging pom from ClassRealm[plexus.core, parent: null]
[DEBUG] Extension realms for project org.springframework.boot:spring-boot-dependencies:pom:2.2.4.RELEASE: (none)
[DEBUG] Looking up lifecycle mappings for packaging pom from ClassRealm[plexus.core, parent: null]
[DEBUG] Resolving plugin prefix oc from [org.apache.maven.plugins, org.codehaus.mojo]
[DEBUG] Resolved plugin prefix oc to org.eclipse.jkube:oc-maven-plugin from POM com.ibm.gts.aro:sample:jar:0.0.1-SNAPSHOT
[DEBUG] === REACTOR BUILD PLAN ================================================
[DEBUG] Project: com.ibm.gts.aro:sample:jar:0.0.1-SNAPSHOT
[DEBUG] Tasks:   [oc:build]
[DEBUG] Style:   Regular
[DEBUG] =======================================================================
[INFO] 
[INFO] -----------------------< com.ibm.gts.aro:sample >-----------------------
[INFO] Building sample 0.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[DEBUG] Resolving plugin prefix oc from [org.apache.maven.plugins, org.codehaus.mojo]
[DEBUG] Resolved plugin prefix oc to org.eclipse.jkube:oc-maven-plugin from POM com.ibm.gts.aro:sample:jar:0.0.1-SNAPSHOT
[DEBUG] Lifecycle default -> [validate, initialize, generate-sources, process-sources, generate-resources, process-resources, compile, process-classes, generate-test-sources, process-test-sources, generate-test-resources, process-test-resources, test-compile, process-test-classes, test, prepare-package, package, pre-integration-test, integration-test, post-integration-test, verify, install, deploy]
[DEBUG] Lifecycle clean -> [pre-clean, clean, post-clean]
[DEBUG] Lifecycle site -> [pre-site, site, post-site, site-deploy]
[DEBUG] === PROJECT BUILD PLAN ================================================
[DEBUG] Project:       com.ibm.gts.aro:sample:0.0.1-SNAPSHOT
[DEBUG] Dependencies (collect): []
[DEBUG] Dependencies (resolve): [compile]
[DEBUG] Repositories (dependencies): [central (https://repo.maven.apache.org/maven2, default, releases)]
[DEBUG] Repositories (plugins)     : [central (https://repo.maven.apache.org/maven2, default, releases)]
[DEBUG] -----------------------------------------------------------------------
[DEBUG] Goal:          org.eclipse.jkube:oc-maven-plugin:0.1.1:build (default-cli)
[DEBUG] Style:         Regular
[DEBUG] Configuration: <?xml version="1.0" encoding="UTF-8"?>
<configuration>
  <apiVersion>${docker.apiVersion}</apiVersion>
  <autoCreateCustomNetworks default-value="false">${docker.autoCreateCustomNetworks}</autoCreateCustomNetworks>
  <autoPull>${docker.autoPull}</autoPull>
  <buildRecreate default-value="none">${jkube.build.recreate}</buildRecreate>
  <buildStrategy>${jkube.build.strategy}</buildStrategy>
  <certPath>${docker.certPath}</certPath>
  <containerNamePattern>${docker.containerNamePattern}</containerNamePattern>
  <dockerHost>${docker.host}</dockerHost>
  <environment>${jkube.environment}</environment>
  <filter>${docker.filter}</filter>
  <forcePull default-value="false">${jkube.build.forcePull}</forcePull>
  <generator>
    <includes>
      <include>spring-boot</include>
    </includes>
    <config>
      <spring-boot>
        <alias>spring-boot</alias>
        <from>registry.access.redhat.com/redhat-openjdk-18/openjdk18-openshift:latest</from>
      </spring-boot>
    </config>
  </generator>
  <imagePullPolicy>${docker.imagePullPolicy}</imagePullPolicy>
  <keepContainer default-value="false">${docker.keepContainer}</keepContainer>
  <keepRunning default-value="false">${docker.keepRunning}</keepRunning>
  <logDate>${docker.logDate}</logDate>
  <logStdout default-value="false">${docker.logStdout}</logStdout>
  <maxConnections default-value="100">${docker.maxConnections}</maxConnections>
  <mode>${jkube.mode}</mode>
  <openshiftPullSecret default-value="pullsecret-jkube">${jkube.build.pullSecret}</openshiftPullSecret>
  <outputDirectory default-value="target/docker">${docker.target.dir}</outputDirectory>
  <profile>${jkube.profile}</profile>
  <project default-value="${project}"/>
  <pullRegistry>${docker.pull.registry}</pullRegistry>
  <reactorProjects default-value="${reactorProjects}"/>
  <registry>${docker.registry}</registry>
  <removeVolumes default-value="false">${docker.removeVolumes}</removeVolumes>
  <resourceDir default-value="${basedir}/src/main/jkube">${jkube.resourceDir}</resourceDir>
  <s2iBuildNameSuffix default-value="">${jkube.s2i.buildNameSuffix}</s2iBuildNameSuffix>
  <s2iImageStreamLookupPolicyLocal default-value="true">${jkube.s2i.imageStreamLookupPolicyLocal}</s2iImageStreamLookupPolicyLocal>
  <session default-value="${session}"/>
  <settings default-value="${settings}"/>
  <skip default-value="false">${jkube.skip}</skip>
  <skipBuild default-value="false">${docker.skip.build}</skipBuild>
  <skipBuildPom>${jkube.skip.build.pom}</skipBuildPom>
  <skipExtendedAuth default-value="false">${docker.skip.extendedAuth}</skipExtendedAuth>
  <skipMachine default-value="false">${docker.skip.machine}</skipMachine>
  <sourceDirectory default-value="src/main/docker">${docker.source.dir}</sourceDirectory>
  <useColor default-value="true">${docker.useColor}</useColor>
  <useProjectClasspath default-value="false">${jkube.useProjectClasspath}</useProjectClasspath>
  <verbose default-value="false">${docker.verbose}</verbose>
  <watchInterval default-value="5000">${docker.watchInterval}</watchInterval>
  <watchMode default-value="both">${docker.watchMode}</watchMode>
  <watchPostExec>${docker.watchPostExec}</watchPostExec>
  <watchPostGoal>${docker.watchPostGoal}</watchPostGoal>
</configuration>
[DEBUG] =======================================================================
[DEBUG] Dependency collection stats: {ConflictMarker.analyzeTime=1749248, ConflictMarker.markTime=987805, ConflictMarker.nodeCount=257, ConflictIdSorter.graphTime=1081454, ConflictIdSorter.topsortTime=3924860, ConflictIdSorter.conflictIdCount=103, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=18123712, ConflictResolver.conflictItemCount=189, DefaultDependencyCollector.collectTime=592281961, DefaultDependencyCollector.transformTime=28285458}
[DEBUG] com.ibm.gts.aro:sample:jar:0.0.1-SNAPSHOT
[DEBUG]    org.springframework.boot:spring-boot-starter-actuator:jar:2.2.4.RELEASE:compile
[DEBUG]       org.springframework.boot:spring-boot-starter:jar:2.2.4.RELEASE:compile (version managed from 2.2.4.RELEASE)
[DEBUG]          org.springframework.boot:spring-boot-starter-logging:jar:2.2.4.RELEASE:compile (version managed from 2.2.4.RELEASE)
[DEBUG]             ch.qos.logback:logback-classic:jar:1.2.3:compile (version managed from 1.2.3)
[DEBUG]                ch.qos.logback:logback-core:jar:1.2.3:compile (version managed from 1.2.3)
[DEBUG]             org.apache.logging.log4j:log4j-to-slf4j:jar:2.12.1:compile (version managed from 2.12.1)
[DEBUG]                org.apache.logging.log4j:log4j-api:jar:2.12.1:compile (version managed from 2.12.1)
[DEBUG]             org.slf4j:jul-to-slf4j:jar:1.7.30:compile (version managed from 1.7.30)
[DEBUG]          jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile (version managed from 1.3.5)
[DEBUG]          org.yaml:snakeyaml:jar:1.25:runtime (version managed from 1.25)
[DEBUG]       org.springframework.boot:spring-boot-actuator-autoconfigure:jar:2.2.4.RELEASE:compile (version managed from 2.2.4.RELEASE)
[DEBUG]          org.springframework.boot:spring-boot-actuator:jar:2.2.4.RELEASE:compile (version managed from 2.2.4.RELEASE)
[DEBUG]          com.fasterxml.jackson.core:jackson-databind:jar:2.10.2:compile (version managed from 2.10.2)
[DEBUG]             com.fasterxml.jackson.core:jackson-annotations:jar:2.10.2:compile (version managed from 2.10.2)
[DEBUG]          org.springframework:spring-context:jar:5.2.3.RELEASE:compile (version managed from 5.2.3.RELEASE)
[DEBUG]          com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.10.2:compile (version managed from 2.10.2)
[DEBUG]       io.micrometer:micrometer-core:jar:1.3.2:compile (version managed from 1.3.2)
[DEBUG]          org.hdrhistogram:HdrHistogram:jar:2.1.11:compile
[DEBUG]          org.latencyutils:LatencyUtils:jar:2.0.3:compile
[DEBUG]    org.springframework.boot:spring-boot-starter-thymeleaf:jar:2.2.4.RELEASE:compile
[DEBUG]       org.thymeleaf:thymeleaf-spring5:jar:3.0.11.RELEASE:compile (version managed from 3.0.11.RELEASE)
[DEBUG]          org.thymeleaf:thymeleaf:jar:3.0.11.RELEASE:compile (version managed from 3.0.11.RELEASE)
[DEBUG]             org.attoparser:attoparser:jar:2.0.5.RELEASE:compile
[DEBUG]             org.unbescape:unbescape:jar:1.1.6.RELEASE:compile
[DEBUG]       org.thymeleaf.extras:thymeleaf-extras-java8time:jar:3.0.4.RELEASE:compile (version managed from 3.0.4.RELEASE)
[DEBUG]    org.springframework.boot:spring-boot-starter-web:jar:2.2.4.RELEASE:compile
[DEBUG]       org.springframework.boot:spring-boot-starter-json:jar:2.2.4.RELEASE:compile (version managed from 2.2.4.RELEASE)
[DEBUG]          com.fasterxml.jackson.datatype:jackson-datatype-jdk8:jar:2.10.2:compile (version managed from 2.10.2)
[DEBUG]          com.fasterxml.jackson.module:jackson-module-parameter-names:jar:2.10.2:compile (version managed from 2.10.2)
[DEBUG]       org.springframework.boot:spring-boot-starter-tomcat:jar:2.2.4.RELEASE:compile (version managed from 2.2.4.RELEASE)
[DEBUG]          org.apache.tomcat.embed:tomcat-embed-core:jar:9.0.30:compile (version managed from 9.0.30)
[DEBUG]          org.apache.tomcat.embed:tomcat-embed-el:jar:9.0.30:compile (version managed from 9.0.30)
[DEBUG]          org.apache.tomcat.embed:tomcat-embed-websocket:jar:9.0.30:compile (version managed from 9.0.30)
[DEBUG]       org.springframework.boot:spring-boot-starter-validation:jar:2.2.4.RELEASE:compile (version managed from 2.2.4.RELEASE)
[DEBUG]          jakarta.validation:jakarta.validation-api:jar:2.0.2:compile (version managed from 2.0.2)
[DEBUG]          org.hibernate.validator:hibernate-validator:jar:6.0.18.Final:compile (version managed from 6.0.18.Final)
[DEBUG]             org.jboss.logging:jboss-logging:jar:3.4.1.Final:compile (version managed from 3.3.2.Final)
[DEBUG]             com.fasterxml:classmate:jar:1.5.1:compile (version managed from 1.3.4)
[DEBUG]       org.springframework:spring-web:jar:5.2.3.RELEASE:compile (version managed from 5.2.3.RELEASE)
[DEBUG]          org.springframework:spring-beans:jar:5.2.3.RELEASE:compile (version managed from 5.2.3.RELEASE)
[DEBUG]       org.springframework:spring-webmvc:jar:5.2.3.RELEASE:compile (version managed from 5.2.3.RELEASE)
[DEBUG]          org.springframework:spring-aop:jar:5.2.3.RELEASE:compile (version managed from 5.2.3.RELEASE)
[DEBUG]          org.springframework:spring-expression:jar:5.2.3.RELEASE:compile (version managed from 5.2.3.RELEASE)
[DEBUG]    org.webjars:webjars-locator-core:jar:0.41:compile
[DEBUG]       org.slf4j:slf4j-api:jar:1.7.30:compile (version managed from 1.7.7)
[DEBUG]       io.github.classgraph:classgraph:jar:4.8.44:compile
[DEBUG]       com.fasterxml.jackson.core:jackson-core:jar:2.10.2:compile (version managed from 2.9.8)
[DEBUG]       org.webjars.npm:angular__http:jar:2.4.10:compile
[DEBUG]    org.webjars:bootstrap:jar:3.3.7:compile
[DEBUG]    org.webjars:jquery:jar:3.1.1-1:compile
[DEBUG]    org.springframework.boot:spring-boot-devtools:jar:2.2.4.RELEASE:runtime (optional)
[DEBUG]       org.springframework.boot:spring-boot:jar:2.2.4.RELEASE:compile (version managed from 2.2.4.RELEASE)
[DEBUG]       org.springframework.boot:spring-boot-autoconfigure:jar:2.2.4.RELEASE:compile (version managed from 2.2.4.RELEASE)
[DEBUG]    org.springframework.boot:spring-boot-starter-test:jar:2.2.4.RELEASE:test
[DEBUG]       org.springframework.boot:spring-boot-test:jar:2.2.4.RELEASE:test (version managed from 2.2.4.RELEASE)
[DEBUG]       org.springframework.boot:spring-boot-test-autoconfigure:jar:2.2.4.RELEASE:test (version managed from 2.2.4.RELEASE)
[DEBUG]       com.jayway.jsonpath:json-path:jar:2.4.0:test (version managed from 2.4.0)
[DEBUG]          net.minidev:json-smart:jar:2.3:test
[DEBUG]             net.minidev:accessors-smart:jar:1.2:test
[DEBUG]                org.ow2.asm:asm:jar:5.0.4:test
[DEBUG]       jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:test (version managed from 2.3.2)
[DEBUG]          jakarta.activation:jakarta.activation-api:jar:1.2.1:test (version managed from 1.2.1)
[DEBUG]       org.junit.jupiter:junit-jupiter:jar:5.5.2:test (version managed from 5.5.2)
[DEBUG]          org.junit.jupiter:junit-jupiter-api:jar:5.5.2:test (version managed from 5.5.2)
[DEBUG]             org.apiguardian:apiguardian-api:jar:1.1.0:test
[DEBUG]             org.opentest4j:opentest4j:jar:1.2.0:test
[DEBUG]             org.junit.platform:junit-platform-commons:jar:1.5.2:test (version managed from 1.5.2)
[DEBUG]          org.junit.jupiter:junit-jupiter-params:jar:5.5.2:test (version managed from 5.5.2)
[DEBUG]          org.junit.jupiter:junit-jupiter-engine:jar:5.5.2:test (version managed from 5.5.2)
[DEBUG]             org.junit.platform:junit-platform-engine:jar:1.5.2:test (version managed from 1.5.2)
[DEBUG]       org.mockito:mockito-junit-jupiter:jar:3.1.0:test (version managed from 3.1.0)
[DEBUG]       org.assertj:assertj-core:jar:3.13.2:test (version managed from 3.13.2)
[DEBUG]       org.hamcrest:hamcrest:jar:2.1:test (version managed from 2.1)
[DEBUG]       org.mockito:mockito-core:jar:3.1.0:test (version managed from 3.1.0)
[DEBUG]          net.bytebuddy:byte-buddy:jar:1.10.6:test (version managed from 1.9.10)
[DEBUG]          net.bytebuddy:byte-buddy-agent:jar:1.10.6:test (version managed from 1.9.10)
[DEBUG]          org.objenesis:objenesis:jar:2.6:test
[DEBUG]       org.skyscreamer:jsonassert:jar:1.5.0:test (version managed from 1.5.0)
[DEBUG]          com.vaadin.external.google:android-json:jar:0.0.20131108.vaadin1:test
[DEBUG]       org.springframework:spring-core:jar:5.2.3.RELEASE:compile (version managed from 5.2.3.RELEASE)
[DEBUG]          org.springframework:spring-jcl:jar:5.2.3.RELEASE:compile (version managed from 5.2.3.RELEASE)
[DEBUG]       org.springframework:spring-test:jar:5.2.3.RELEASE:test (version managed from 5.2.3.RELEASE)
[DEBUG]       org.xmlunit:xmlunit-core:jar:2.6.3:test (version managed from 2.6.3)
[DEBUG]    org.springframework.boot:spring-boot-starter-data-redis:jar:2.2.4.RELEASE:compile
[DEBUG]       org.springframework.data:spring-data-redis:jar:2.2.4.RELEASE:compile (version managed from 2.2.4.RELEASE)
[DEBUG]          org.springframework.data:spring-data-keyvalue:jar:2.2.4.RELEASE:compile (version managed from 2.2.4.RELEASE)
[DEBUG]             org.springframework.data:spring-data-commons:jar:2.2.4.RELEASE:compile (version managed from 2.2.4.RELEASE)
[DEBUG]          org.springframework:spring-tx:jar:5.2.3.RELEASE:compile (version managed from 5.2.3.RELEASE)
[DEBUG]          org.springframework:spring-oxm:jar:5.2.3.RELEASE:compile (version managed from 5.2.3.RELEASE)
[DEBUG]          org.springframework:spring-context-support:jar:5.2.3.RELEASE:compile (version managed from 5.2.3.RELEASE)
[DEBUG]       io.lettuce:lettuce-core:jar:5.2.1.RELEASE:compile (version managed from 5.2.1.RELEASE)
[DEBUG]          io.netty:netty-common:jar:4.1.45.Final:compile (version managed from 4.1.43.Final)
[DEBUG]          io.netty:netty-handler:jar:4.1.45.Final:compile (version managed from 4.1.43.Final)
[DEBUG]             io.netty:netty-buffer:jar:4.1.45.Final:compile (version managed from 4.1.45.Final)
[DEBUG]             io.netty:netty-codec:jar:4.1.45.Final:compile (version managed from 4.1.45.Final)
[DEBUG]          io.netty:netty-transport:jar:4.1.45.Final:compile (version managed from 4.1.43.Final)
[DEBUG]             io.netty:netty-resolver:jar:4.1.45.Final:compile (version managed from 4.1.45.Final)
[DEBUG]          io.projectreactor:reactor-core:jar:3.3.2.RELEASE:compile (version managed from 3.3.0.RELEASE)
[DEBUG]             org.reactivestreams:reactive-streams:jar:1.0.3:compile (version managed from 1.0.3)
[DEBUG]    org.projectlombok:lombok:jar:1.18.10:compile
[INFO] 
[INFO] --- oc-maven-plugin:0.1.1:build (default-cli) @ sample ---
[DEBUG] Dependency collection stats: {ConflictMarker.analyzeTime=347781, ConflictMarker.markTime=281258, ConflictMarker.nodeCount=224, ConflictIdSorter.graphTime=361273, ConflictIdSorter.topsortTime=133903, ConflictIdSorter.conflictIdCount=132, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=8676180, ConflictResolver.conflictItemCount=199, DefaultDependencyCollector.collectTime=735910120, DefaultDependencyCollector.transformTime=9955784}
[DEBUG] org.eclipse.jkube:oc-maven-plugin:jar:0.1.1:
[DEBUG]    org.eclipse.jkube:jkube-kit-build-service-docker:jar:0.1.1:compile
[DEBUG]       org.eclipse.jkube:jkube-kit-build-maven:jar:0.1.1:compile
[DEBUG]          org.eclipse.jkube:jkube-kit-build-api:jar:0.1.1:compile
[DEBUG]             org.eclipse.jkube:jkube-kit-config-image:jar:0.1.1:compile
[DEBUG]             commons-codec:commons-codec:jar:1.13:compile
[DEBUG]       org.apache.httpcomponents:httpclient:jar:4.5.6:compile
[DEBUG]          org.apache.httpcomponents:httpcore:jar:4.4.10:compile
[DEBUG]          commons-logging:commons-logging:jar:1.2:compile
[DEBUG]       com.github.jnr:jnr-unixsocket:jar:0.12:compile
[DEBUG]          com.github.jnr:jnr-ffi:jar:2.0.9:compile
[DEBUG]             com.github.jnr:jffi:jar:1.2.11:compile
[DEBUG]             com.github.jnr:jffi:jar:native:1.2.11:runtime
[DEBUG]             org.ow2.asm:asm:jar:5.0.3:compile
[DEBUG]             org.ow2.asm:asm-commons:jar:5.0.3:compile
[DEBUG]             org.ow2.asm:asm-analysis:jar:5.0.3:compile
[DEBUG]             org.ow2.asm:asm-tree:jar:5.0.3:compile
[DEBUG]             org.ow2.asm:asm-util:jar:5.0.3:compile
[DEBUG]             com.github.jnr:jnr-x86asm:jar:1.0.2:compile
[DEBUG]          com.github.jnr:jnr-constants:jar:0.9.1:compile
[DEBUG]          com.github.jnr:jnr-enxio:jar:0.12:compile
[DEBUG]          com.github.jnr:jnr-posix:jar:3.0.29:compile
[DEBUG]       org.bouncycastle:bcpkix-jdk15on:jar:1.61:compile
[DEBUG]          org.bouncycastle:bcprov-jdk15on:jar:1.61:compile
[DEBUG]       org.apache.commons:commons-text:jar:1.1:compile
[DEBUG]    org.eclipse.jkube:jkube-kit-enricher-generic:jar:0.1.1:compile
[DEBUG]       org.eclipse.jkube:jkube-kit-enricher-api:jar:0.1.1:compile
[DEBUG]          org.eclipse.jkube:jkube-kit-resource-service:jar:0.1.1:compile
[DEBUG]          com.google.code.findbugs:jsr305:jar:3.0.2:compile
[DEBUG]       io.fabric8:ianaservicehelper:jar:0.0.1:compile
[DEBUG]       com.google.code.gson:gson:jar:2.8.5:compile
[DEBUG]    org.eclipse.jkube:jkube-kit-enricher-specific:jar:0.1.1:compile
[DEBUG]    org.eclipse.jkube:jkube-kit-generator-java-exec:jar:0.1.1:compile
[DEBUG]       org.eclipse.jkube:jkube-kit-generator-api:jar:0.1.1:compile
[DEBUG]          org.jboss.shrinkwrap:shrinkwrap-api:jar:1.2.6:compile
[DEBUG]          org.jboss.shrinkwrap:shrinkwrap-spi:jar:1.2.6:compile
[DEBUG]          org.jboss.shrinkwrap:shrinkwrap-impl-base:jar:1.2.6:compile
[DEBUG]    org.eclipse.jkube:jkube-kit-generator-karaf:jar:0.1.1:compile
[DEBUG]    org.eclipse.jkube:jkube-kit-generator-webapp:jar:0.1.1:compile
[DEBUG]    org.eclipse.jkube:jkube-kit-generator-wildfly-swarm:jar:0.1.1:compile
[DEBUG]    org.eclipse.jkube:jkube-kit-thorntail:jar:0.1.1:compile
[DEBUG]    org.eclipse.jkube:jkube-kit-spring-boot:jar:0.1.1:compile
[DEBUG]       org.eclipse.jkube:jkube-kit-watcher-api:jar:0.1.1:compile
[DEBUG]    org.eclipse.jkube:jkube-kit-quarkus:jar:0.1.1:compile
[DEBUG]       org.eclipse.jkube:jkube-kit-common-maven:jar:0.1.1:compile
[DEBUG]          org.eclipse.jkube:jkube-kit-common:jar:0.1.1:compile
[DEBUG]             io.fabric8:kubernetes-client:jar:4.6.2:compile
[DEBUG]                io.fabric8:kubernetes-model:jar:4.6.2:compile
[DEBUG]                   io.fabric8:kubernetes-model-common:jar:4.6.2:compile
[DEBUG]                   com.fasterxml.jackson.module:jackson-module-jaxb-annotations:jar:2.9.10:compile
[DEBUG]                com.squareup.okhttp3:okhttp:jar:3.12.6:compile
[DEBUG]                   com.squareup.okio:okio:jar:1.15.0:compile
[DEBUG]                com.squareup.okhttp3:logging-interceptor:jar:3.12.6:compile
[DEBUG]                io.fabric8:zjsonpatch:jar:0.3.0:compile
[DEBUG]                com.github.mifmif:generex:jar:1.0.2:compile
[DEBUG]                   dk.brics.automaton:automaton:jar:1.11-8:compile
[DEBUG]             io.fabric8:openshift-client:jar:4.6.2:compile
[DEBUG]             org.apache.commons:commons-lang3:jar:3.8.1:compile
[DEBUG]             org.apache.commons:commons-compress:jar:1.19:compile
[DEBUG]             commons-io:commons-io:jar:2.6:compile
[DEBUG]             com.google.guava:guava:jar:27.0-jre:compile
[DEBUG]                com.google.guava:failureaccess:jar:1.0:compile
[DEBUG]                com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[DEBUG]                org.checkerframework:checker-qual:jar:2.5.2:compile
[DEBUG]                com.google.errorprone:error_prone_annotations:jar:2.2.0:compile
[DEBUG]                com.google.j2objc:j2objc-annotations:jar:1.1:compile
[DEBUG]                org.codehaus.mojo:animal-sniffer-annotations:jar:1.17:compile
[DEBUG]             com.fasterxml.jackson.core:jackson-core:jar:2.10.0:compile
[DEBUG]             com.fasterxml.jackson.core:jackson-databind:jar:2.10.0:compile
[DEBUG]                com.fasterxml.jackson.core:jackson-annotations:jar:2.10.0:compile
[DEBUG]             com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.10.0:compile
[DEBUG]             org.slf4j:slf4j-api:jar:1.7.25:compile
[DEBUG]             org.yaml:snakeyaml:jar:1.25:compile
[DEBUG]             org.eclipse.jgit:org.eclipse.jgit:jar:5.3.0.201903130848-r:compile
[DEBUG]                com.jcraft:jsch:jar:0.1.54:compile
[DEBUG]                com.jcraft:jzlib:jar:1.1.1:compile
[DEBUG]                com.googlecode.javaewah:JavaEWAH:jar:1.1.6:compile
[DEBUG]                org.bouncycastle:bcpg-jdk15on:jar:1.60:compile
[DEBUG]             javax.validation:validation-api:jar:2.0.1.Final:compile
[DEBUG]             org.fusesource.jansi:jansi:jar:1.16:compile
[DEBUG]             org.javassist:javassist:jar:3.20.0-GA:compile
[DEBUG]             com.networknt:json-schema-validator:jar:0.1.7:compile
[DEBUG]                org.slf4j:slf4j-ext:jar:1.7.25:compile
[DEBUG]          org.apache.maven:maven-core:jar:3.5.4:compile
[DEBUG]             org.apache.maven:maven-model:jar:3.5.4:compile
[DEBUG]             org.apache.maven:maven-settings:jar:3.5.4:compile
[DEBUG]             org.apache.maven:maven-settings-builder:jar:3.5.4:compile
[DEBUG]                org.codehaus.plexus:plexus-interpolation:jar:1.24:compile
[DEBUG]                org.sonatype.plexus:plexus-sec-dispatcher:jar:1.4:compile
[DEBUG]                   org.sonatype.plexus:plexus-cipher:jar:1.4:compile
[DEBUG]             org.apache.maven:maven-builder-support:jar:3.5.4:compile
[DEBUG]             org.apache.maven:maven-repository-metadata:jar:3.5.4:compile
[DEBUG]             org.apache.maven:maven-artifact:jar:3.5.4:compile
[DEBUG]             org.apache.maven:maven-plugin-api:jar:3.5.4:compile
[DEBUG]             org.apache.maven:maven-model-builder:jar:3.5.4:compile
[DEBUG]             org.apache.maven:maven-resolver-provider:jar:3.5.4:compile
[DEBUG]             org.apache.maven.resolver:maven-resolver-impl:jar:1.1.1:compile
[DEBUG]             org.apache.maven.resolver:maven-resolver-api:jar:1.1.1:compile
[DEBUG]             org.apache.maven.resolver:maven-resolver-spi:jar:1.1.1:compile
[DEBUG]             org.apache.maven.resolver:maven-resolver-util:jar:1.1.1:compile
[DEBUG]             org.apache.maven.shared:maven-shared-utils:jar:3.2.1:compile
[DEBUG]             org.eclipse.sisu:org.eclipse.sisu.plexus:jar:0.3.3:compile
[DEBUG]                javax.enterprise:cdi-api:jar:1.0:compile
[DEBUG]                   javax.annotation:jsr250-api:jar:1.0:compile
[DEBUG]             org.eclipse.sisu:org.eclipse.sisu.inject:jar:0.3.3:compile
[DEBUG]             com.google.inject:guice:jar:no_aop:4.2.0:compile
[DEBUG]                aopalliance:aopalliance:jar:1.0:compile
[DEBUG]             javax.inject:javax.inject:jar:1:compile
[DEBUG]             org.codehaus.plexus:plexus-classworlds:jar:2.5.2:compile
[DEBUG]          org.codehaus.plexus:plexus-archiver:jar:3.5:compile
[DEBUG]             org.codehaus.plexus:plexus-utils:jar:3.0.24:compile
[DEBUG]             org.codehaus.plexus:plexus-io:jar:3.0.0:compile
[DEBUG]             org.iq80.snappy:snappy:jar:0.4:compile
[DEBUG]             org.tukaani:xz:jar:1.6:runtime
[DEBUG]          org.apache.maven.plugins:maven-assembly-plugin:jar:3.1.1:compile
[DEBUG]             org.apache.maven.shared:maven-artifact-transfer:jar:0.9.0:compile
[DEBUG]             org.apache.maven.shared:file-management:jar:3.0.0:compile
[DEBUG]             org.apache.maven.shared:maven-shared-io:jar:3.0.0:compile
[DEBUG]                org.apache.maven.wagon:wagon-provider-api:jar:2.10:compile
[DEBUG]             org.apache.maven.shared:maven-filtering:jar:3.1.1:compile
[DEBUG]                org.sonatype.plexus:plexus-build-api:jar:0.0.7:compile
[DEBUG]             org.apache.maven:maven-archiver:jar:3.2.0:compile
[DEBUG]          org.apache.maven.shared:maven-common-artifact-filters:jar:3.1.0:compile
[DEBUG]             org.sonatype.sisu:sisu-inject-plexus:jar:1.4.2:compile
[DEBUG]                org.codehaus.plexus:plexus-component-annotations:jar:1.5.4:compile
[DEBUG]                org.sonatype.sisu:sisu-inject-bean:jar:1.4.2:compile
[DEBUG]                   org.sonatype.sisu:sisu-guice:jar:noaop:2.1.7:compile
[DEBUG]    org.eclipse.jkube:jkube-kit-vertx:jar:0.1.1:compile
[DEBUG]    org.eclipse.jkube:jkube-kit-profiles:jar:0.1.1:compile
[DEBUG]       org.eclipse.jkube:jkube-kit-config-resource:jar:0.1.1:compile
[DEBUG]          org.apache.maven.plugin-tools:maven-plugin-annotations:jar:3.6.0:compile
[DEBUG]    org.eclipse.jkube:jkube-kit-watcher-standard:jar:0.1.1:compile
[DEBUG] Created new class realm plugin>org.eclipse.jkube:oc-maven-plugin:0.1.1
[DEBUG] Importing foreign packages into class realm plugin>org.eclipse.jkube:oc-maven-plugin:0.1.1
[DEBUG]   Imported:  < maven.api
[DEBUG] Populating class realm plugin>org.eclipse.jkube:oc-maven-plugin:0.1.1
[DEBUG]   Included: org.eclipse.jkube:oc-maven-plugin:jar:0.1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-build-service-docker:jar:0.1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-build-maven:jar:0.1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-build-api:jar:0.1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-config-image:jar:0.1.1
[DEBUG]   Included: commons-codec:commons-codec:jar:1.13
[DEBUG]   Included: org.apache.httpcomponents:httpclient:jar:4.5.6
[DEBUG]   Included: org.apache.httpcomponents:httpcore:jar:4.4.10
[DEBUG]   Included: commons-logging:commons-logging:jar:1.2
[DEBUG]   Included: com.github.jnr:jnr-unixsocket:jar:0.12
[DEBUG]   Included: com.github.jnr:jnr-ffi:jar:2.0.9
[DEBUG]   Included: com.github.jnr:jffi:jar:1.2.11
[DEBUG]   Included: com.github.jnr:jffi:jar:native:1.2.11
[DEBUG]   Included: org.ow2.asm:asm:jar:5.0.3
[DEBUG]   Included: org.ow2.asm:asm-commons:jar:5.0.3
[DEBUG]   Included: org.ow2.asm:asm-analysis:jar:5.0.3
[DEBUG]   Included: org.ow2.asm:asm-tree:jar:5.0.3
[DEBUG]   Included: org.ow2.asm:asm-util:jar:5.0.3
[DEBUG]   Included: com.github.jnr:jnr-x86asm:jar:1.0.2
[DEBUG]   Included: com.github.jnr:jnr-constants:jar:0.9.1
[DEBUG]   Included: com.github.jnr:jnr-enxio:jar:0.12
[DEBUG]   Included: com.github.jnr:jnr-posix:jar:3.0.29
[DEBUG]   Included: org.bouncycastle:bcpkix-jdk15on:jar:1.61
[DEBUG]   Included: org.bouncycastle:bcprov-jdk15on:jar:1.61
[DEBUG]   Included: org.apache.commons:commons-text:jar:1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-enricher-generic:jar:0.1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-enricher-api:jar:0.1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-resource-service:jar:0.1.1
[DEBUG]   Included: com.google.code.findbugs:jsr305:jar:3.0.2
[DEBUG]   Included: io.fabric8:ianaservicehelper:jar:0.0.1
[DEBUG]   Included: com.google.code.gson:gson:jar:2.8.5
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-enricher-specific:jar:0.1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-generator-java-exec:jar:0.1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-generator-api:jar:0.1.1
[DEBUG]   Included: org.jboss.shrinkwrap:shrinkwrap-api:jar:1.2.6
[DEBUG]   Included: org.jboss.shrinkwrap:shrinkwrap-spi:jar:1.2.6
[DEBUG]   Included: org.jboss.shrinkwrap:shrinkwrap-impl-base:jar:1.2.6
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-generator-karaf:jar:0.1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-generator-webapp:jar:0.1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-generator-wildfly-swarm:jar:0.1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-thorntail:jar:0.1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-spring-boot:jar:0.1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-watcher-api:jar:0.1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-quarkus:jar:0.1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-common-maven:jar:0.1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-common:jar:0.1.1
[DEBUG]   Included: io.fabric8:kubernetes-client:jar:4.6.2
[DEBUG]   Included: io.fabric8:kubernetes-model:jar:4.6.2
[DEBUG]   Included: io.fabric8:kubernetes-model-common:jar:4.6.2
[DEBUG]   Included: com.fasterxml.jackson.module:jackson-module-jaxb-annotations:jar:2.9.10
[DEBUG]   Included: com.squareup.okhttp3:okhttp:jar:3.12.6
[DEBUG]   Included: com.squareup.okio:okio:jar:1.15.0
[DEBUG]   Included: com.squareup.okhttp3:logging-interceptor:jar:3.12.6
[DEBUG]   Included: io.fabric8:zjsonpatch:jar:0.3.0
[DEBUG]   Included: com.github.mifmif:generex:jar:1.0.2
[DEBUG]   Included: dk.brics.automaton:automaton:jar:1.11-8
[DEBUG]   Included: io.fabric8:openshift-client:jar:4.6.2
[DEBUG]   Included: org.apache.commons:commons-lang3:jar:3.8.1
[DEBUG]   Included: org.apache.commons:commons-compress:jar:1.19
[DEBUG]   Included: commons-io:commons-io:jar:2.6
[DEBUG]   Included: com.google.guava:guava:jar:27.0-jre
[DEBUG]   Included: com.google.guava:failureaccess:jar:1.0
[DEBUG]   Included: com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava
[DEBUG]   Included: org.checkerframework:checker-qual:jar:2.5.2
[DEBUG]   Included: com.google.errorprone:error_prone_annotations:jar:2.2.0
[DEBUG]   Included: com.google.j2objc:j2objc-annotations:jar:1.1
[DEBUG]   Included: org.codehaus.mojo:animal-sniffer-annotations:jar:1.17
[DEBUG]   Included: com.fasterxml.jackson.core:jackson-core:jar:2.10.0
[DEBUG]   Included: com.fasterxml.jackson.core:jackson-databind:jar:2.10.0
[DEBUG]   Included: com.fasterxml.jackson.core:jackson-annotations:jar:2.10.0
[DEBUG]   Included: com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.10.0
[DEBUG]   Included: org.yaml:snakeyaml:jar:1.25
[DEBUG]   Included: org.eclipse.jgit:org.eclipse.jgit:jar:5.3.0.201903130848-r
[DEBUG]   Included: com.jcraft:jsch:jar:0.1.54
[DEBUG]   Included: com.jcraft:jzlib:jar:1.1.1
[DEBUG]   Included: com.googlecode.javaewah:JavaEWAH:jar:1.1.6
[DEBUG]   Included: org.bouncycastle:bcpg-jdk15on:jar:1.60
[DEBUG]   Included: javax.validation:validation-api:jar:2.0.1.Final
[DEBUG]   Included: org.javassist:javassist:jar:3.20.0-GA
[DEBUG]   Included: com.networknt:json-schema-validator:jar:0.1.7
[DEBUG]   Included: org.slf4j:slf4j-ext:jar:1.7.25
[DEBUG]   Included: org.codehaus.plexus:plexus-interpolation:jar:1.24
[DEBUG]   Included: org.sonatype.plexus:plexus-sec-dispatcher:jar:1.4
[DEBUG]   Included: org.sonatype.plexus:plexus-cipher:jar:1.4
[DEBUG]   Included: org.apache.maven:maven-builder-support:jar:3.5.4
[DEBUG]   Included: org.apache.maven.resolver:maven-resolver-util:jar:1.1.1
[DEBUG]   Included: org.apache.maven.shared:maven-shared-utils:jar:3.2.1
[DEBUG]   Included: javax.enterprise:cdi-api:jar:1.0
[DEBUG]   Included: org.eclipse.sisu:org.eclipse.sisu.inject:jar:0.3.3
[DEBUG]   Included: com.google.inject:guice:jar:no_aop:4.2.0
[DEBUG]   Included: aopalliance:aopalliance:jar:1.0
[DEBUG]   Included: org.codehaus.plexus:plexus-archiver:jar:3.5
[DEBUG]   Included: org.codehaus.plexus:plexus-utils:jar:3.0.24
[DEBUG]   Included: org.codehaus.plexus:plexus-io:jar:3.0.0
[DEBUG]   Included: org.iq80.snappy:snappy:jar:0.4
[DEBUG]   Included: org.tukaani:xz:jar:1.6
[DEBUG]   Included: org.apache.maven.plugins:maven-assembly-plugin:jar:3.1.1
[DEBUG]   Included: org.apache.maven.shared:maven-artifact-transfer:jar:0.9.0
[DEBUG]   Included: org.apache.maven.shared:file-management:jar:3.0.0
[DEBUG]   Included: org.apache.maven.shared:maven-shared-io:jar:3.0.0
[DEBUG]   Included: org.apache.maven.shared:maven-filtering:jar:3.1.1
[DEBUG]   Included: org.sonatype.plexus:plexus-build-api:jar:0.0.7
[DEBUG]   Included: org.apache.maven:maven-archiver:jar:3.2.0
[DEBUG]   Included: org.apache.maven.shared:maven-common-artifact-filters:jar:3.1.0
[DEBUG]   Included: org.codehaus.plexus:plexus-component-annotations:jar:1.5.4
[DEBUG]   Included: org.sonatype.sisu:sisu-inject-bean:jar:1.4.2
[DEBUG]   Included: org.sonatype.sisu:sisu-guice:jar:noaop:2.1.7
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-vertx:jar:0.1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-profiles:jar:0.1.1
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-config-resource:jar:0.1.1
[DEBUG]   Included: org.apache.maven.plugin-tools:maven-plugin-annotations:jar:3.6.0
[DEBUG]   Included: org.eclipse.jkube:jkube-kit-watcher-standard:jar:0.1.1
[DEBUG] Configuring mojo org.eclipse.jkube:oc-maven-plugin:0.1.1:build from plugin realm ClassRealm[plugin>org.eclipse.jkube:oc-maven-plugin:0.1.1, parent: sun.misc.Launcher$AppClassLoader@70dea4e]
[DEBUG] Configuring mojo 'org.eclipse.jkube:oc-maven-plugin:0.1.1:build' with basic configurator -->
[DEBUG]   (f) autoCreateCustomNetworks = false
[DEBUG]   (f) buildRecreate = none
[DEBUG]   (f) forcePull = false
[DEBUG]   (f) includes = [spring-boot]
[DEBUG]   (f) config = {spring-boot={alias=spring-boot, from=registry.access.redhat.com/redhat-openjdk-18/openjdk18-openshift:latest}}
[DEBUG]   (f) generator = org.eclipse.jkube.kit.config.resource.ProcessorConfig@1ffcf674
[DEBUG]   (f) keepContainer = false
[DEBUG]   (f) keepRunning = false
[DEBUG]   (f) logStdout = false
[DEBUG]   (f) maxConnections = 100
[DEBUG]   (f) openshiftPullSecret = pullsecret-jkube
[DEBUG]   (f) outputDirectory = target/docker
[DEBUG]   (f) project = MavenProject: com.ibm.gts.aro:sample:0.0.1-SNAPSHOT @ /Users/Marcelo.Borges@ibm.com/IBM/customers/cibc/sample/pom.xml
[DEBUG]   (f) reactorProjects = [MavenProject: com.ibm.gts.aro:sample:0.0.1-SNAPSHOT @ /Users/Marcelo.Borges@ibm.com/IBM/customers/cibc/sample/pom.xml]
[DEBUG]   (f) removeVolumes = false
[DEBUG]   (f) resourceDir = /Users/Marcelo.Borges@ibm.com/IBM/customers/cibc/sample/src/main/jkube
[DEBUG]   (f) s2iImageStreamLookupPolicyLocal = true
[DEBUG]   (f) session = org.apache.maven.execution.MavenSession@7d070ef5
[DEBUG]   (f) settings = org.apache.maven.execution.SettingsAdapter@2e2f720
[DEBUG]   (f) skip = false
[DEBUG]   (f) skipBuild = false
[DEBUG]   (f) skipExtendedAuth = false
[DEBUG]   (f) skipMachine = false
[DEBUG]   (f) sourceDirectory = src/main/docker
[DEBUG]   (f) useColor = true
[DEBUG]   (f) useProjectClasspath = false
[DEBUG]   (f) verbose = false
[DEBUG]   (f) watchInterval = 5000
[DEBUG]   (f) watchMode = both
[DEBUG] -- end configuration --
[DEBUG] Trying to configure client from Kubernetes config...
[DEBUG] Found for Kubernetes config at: [/Users/Marcelo.Borges@ibm.com/.kube/config].
[DEBUG] Trying to configure client from Kubernetes config...
[DEBUG] Found for Kubernetes config at: [/Users/Marcelo.Borges@ibm.com/.kube/config].
[DEBUG] Trying to configure client from Kubernetes config...
[DEBUG] Found for Kubernetes config at: [/Users/Marcelo.Borges@ibm.com/.kube/config].
[INFO] oc: Running in OpenShift mode
[INFO] oc: Using OpenShift build with strategy S2I
[DEBUG] Trying to configure client from Kubernetes config...
[DEBUG] Found for Kubernetes config at: [/Users/Marcelo.Borges@ibm.com/.kube/config].
[DEBUG] Trying to configure client from Kubernetes config...
[DEBUG] Found for Kubernetes config at: [/Users/Marcelo.Borges@ibm.com/.kube/config].
[DEBUG] Trying to configure client from Kubernetes config...
[DEBUG] Found for Kubernetes config at: [/Users/Marcelo.Borges@ibm.com/.kube/config].
[DEBUG] Trying to configure client from Kubernetes config...
[DEBUG] Found for Kubernetes config at: [/Users/Marcelo.Borges@ibm.com/.kube/config].
[INFO] oc: Running generator spring-boot
[DEBUG] readpipe [git, --version],/usr/bin
[DEBUG] readpipe may return 'git version 2.21.1 (Apple Git-122.3)'
[DEBUG] remaining output:

[DEBUG] readpipe [git, config, --system, --edit],/usr/bin
[DEBUG] readpipe may return '/private/etc/gitconfig'
[DEBUG] remaining output:

[INFO] oc: spring-boot: Using Docker image registry.access.redhat.com/redhat-openjdk-18/openjdk18-openshift:latest as base / builder
[DEBUG] Trying to configure client from Kubernetes config...
[DEBUG] Found for Kubernetes config at: [/Users/Marcelo.Borges@ibm.com/.kube/config].
[DEBUG] Trying to configure client from Kubernetes config...
[DEBUG] Found for Kubernetes config at: [/Users/Marcelo.Borges@ibm.com/.kube/config].
[DEBUG] Trying to configure client from Kubernetes config...
[DEBUG] Found for Kubernetes config at: [/Users/Marcelo.Borges@ibm.com/.kube/config].
[DEBUG] Trying to configure client from Kubernetes config...
[DEBUG] Found for Kubernetes config at: [/Users/Marcelo.Borges@ibm.com/.kube/config].
[DEBUG] All known ContainerDescriptorHandler components: [metaInf-spring, plexus, file-aggregator, metaInf-services]
[DEBUG] FileSet[bin/] dir perms: -1 file perms: -1
[DEBUG] The archive base directory is 'null'
[DEBUG] FileSet[] dir perms: -1 file perms: -1
[DEBUG] The archive base directory is 'null'
[DEBUG] FileSet[] dir perms: -1 file perms: -1
[DEBUG] The archive base directory is 'null'
[DEBUG] NOT reformatting any files in /Users/Marcelo.Borges@ibm.com/IBM/customers/cibc/sample/target
[DEBUG] Adding file-set in: /Users/Marcelo.Borges@ibm.com/IBM/customers/cibc/sample/target to archive location: ./
[DEBUG] Adding exclude for assembly working-directory: docker/aro/sample/latest/work
File-Set source directory: /Users/Marcelo.Borges@ibm.com/IBM/customers/cibc/sample/target
[INFO] Copying files to /Users/Marcelo.Borges@ibm.com/IBM/customers/cibc/sample/target/docker/aro/sample/latest/build/maven
[INFO] Building tar: /Users/Marcelo.Borges@ibm.com/IBM/customers/cibc/sample/target/docker/aro/sample/latest/tmp/docker-build.tar
[DEBUG] Dockerfile already added, skipping
[INFO] oc: [aro/sample:latest] "spring-boot": Created docker source tar /Users/Marcelo.Borges@ibm.com/IBM/customers/cibc/sample/target/docker/aro/sample/latest/tmp/docker-build.tar
[ERROR] oc: Failed to execute the build [Unable to build the image using the OpenShift build service]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  8.636 s
[INFO] Finished at: 2020-03-04T22:53:40-06:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.jkube:oc-maven-plugin:0.1.1:build (default-cli) on project sample: Failed to execute the build: Unable to build the image using the OpenShift build service: NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.jkube:oc-maven-plugin:0.1.1:build (default-cli) on project sample: Failed to execute the build
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to execute the build
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.buildAndTag (AbstractDockerMojo.java:690)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.processImageConfig (AbstractDockerMojo.java:513)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.executeBuildGoal (AbstractDockerMojo.java:657)
    at org.eclipse.jkube.maven.plugin.mojo.build.BuildMojo.executeInternal (BuildMojo.java:105)
    at org.eclipse.jkube.maven.plugin.mojo.build.BuildMojo.executeDockerBuild (BuildMojo.java:132)
    at org.eclipse.jkube.maven.plugin.mojo.build.BuildMojo.execute (BuildMojo.java:72)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.eclipse.jkube.kit.config.service.JkubeServiceException: Unable to build the image using the OpenShift build service
    at org.eclipse.jkube.kit.config.service.openshift.OpenshiftBuildService.build (OpenshiftBuildService.java:140)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.buildAndTag (AbstractDockerMojo.java:687)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.processImageConfig (AbstractDockerMojo.java:513)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.executeBuildGoal (AbstractDockerMojo.java:657)
    at org.eclipse.jkube.maven.plugin.mojo.build.BuildMojo.executeInternal (BuildMojo.java:105)
    at org.eclipse.jkube.maven.plugin.mojo.build.BuildMojo.executeDockerBuild (BuildMojo.java:132)
    at org.eclipse.jkube.maven.plugin.mojo.build.BuildMojo.execute (BuildMojo.java:72)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.lang.NullPointerException
    at org.eclipse.jkube.kit.config.service.openshift.OpenshiftBuildService.checkOrCreatePullSecret (OpenshiftBuildService.java:388)
    at org.eclipse.jkube.kit.config.service.openshift.OpenshiftBuildService.build (OpenshiftBuildService.java:114)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.buildAndTag (AbstractDockerMojo.java:687)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.processImageConfig (AbstractDockerMojo.java:513)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.executeBuildGoal (AbstractDockerMojo.java:657)
    at org.eclipse.jkube.maven.plugin.mojo.build.BuildMojo.executeInternal (BuildMojo.java:105)
    at org.eclipse.jkube.maven.plugin.mojo.build.BuildMojo.executeDockerBuild (BuildMojo.java:132)
    at org.eclipse.jkube.maven.plugin.mojo.build.BuildMojo.execute (BuildMojo.java:72)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
➜  sample git:(master) ✗ 
```

--- a/jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/service/docker/auth/AuthConfigFactory.java
+++ b/jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/service/docker/auth/AuthConfigFactory.java
@@ -61,19 +61,12 @@ public class AuthConfigFactory {
 
     static final String DOCKER_LOGIN_DEFAULT_REGISTRY = "https://index.docker.io/v1/";
 
-    private KitLogger log;
+    private final KitLogger log;
     private static final String[] DEFAULT_REGISTRIES = new String[]{
             "docker.io", "index.docker.io", "registry.hub.docker.com"
     };
 
-    /**
-     * Constructor which should be used during startup phase of a plugin
-     *
-     */
-    public AuthConfigFactory() {
-    }
-
-    public void setLog(KitLogger log) {
+    public AuthConfigFactory(KitLogger log) {
         this.log = log;
     }
 

--- a/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/AbstractDockerMojo.java
+++ b/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/AbstractDockerMojo.java
@@ -428,7 +428,6 @@ public abstract class AbstractDockerMojo extends AbstractMojo implements ConfigH
     @Override
     public void contextualize(Context context) throws ContextException {
         plexusContainer = ((PlexusContainer) context.get(PlexusConstants.PLEXUS_KEY));
-        authConfigFactory = new AuthConfigFactory();
     }
 
     @Override
@@ -436,13 +435,11 @@ public abstract class AbstractDockerMojo extends AbstractMojo implements ConfigH
         if (!skip) {
             boolean ansiRestore = Ansi.isEnabled();
             log = new AnsiLogger(getLog(), useColorForLogging(), verbose, !settings.getInteractiveMode(), getLogPrefix());
+            authConfigFactory = new AuthConfigFactory(log);
+            imageConfigResolver.setLog(log);
 
             try {
-                authConfigFactory.setLog(log);
-                imageConfigResolver.setLog(log);
-
                 LogOutputSpecFactory logSpecFactory = new LogOutputSpecFactory(useColor, logStdout, logDate);
-
 
                 DockerAccess access = null;
                 try {
@@ -575,7 +572,7 @@ public abstract class AbstractDockerMojo extends AbstractMojo implements ConfigH
         }
     }
 
-    protected RegistryService.RegistryConfig getRegistryConfig(String specificRegistry) throws MojoExecutionException {
+    protected RegistryService.RegistryConfig getRegistryConfig(String specificRegistry) {
         return new RegistryService.RegistryConfig.Builder()
                 .settings(MavenUtil.getRegistryServerFromMavenSettings(settings))
                 .authConfig(authConfig != null ? authConfig.toMap() : null)

--- a/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/BuildMojo.java
+++ b/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/BuildMojo.java
@@ -18,6 +18,7 @@ import org.eclipse.jkube.kit.build.service.docker.DockerAccessFactory;
 import org.eclipse.jkube.kit.build.service.docker.ServiceHub;
 import org.eclipse.jkube.kit.build.service.docker.access.DockerAccess;
 import org.eclipse.jkube.kit.build.service.docker.access.log.LogOutputSpecFactory;
+import org.eclipse.jkube.kit.build.service.docker.auth.AuthConfigFactory;
 import org.eclipse.jkube.kit.build.service.docker.config.ConfigHelper;
 import org.eclipse.jkube.kit.common.util.AnsiLogger;
 import org.eclipse.jkube.kit.common.util.MavenUtil;
@@ -86,7 +87,7 @@ public class BuildMojo extends AbstractDockerMojo implements Contextualizable {
     public void executeDockerBuild() throws MojoExecutionException, MojoFailureException {
         if (!skip) {
             log = new AnsiLogger(getLog(), useColor, verbose, !settings.getInteractiveMode(), getLogPrefix());
-            authConfigFactory.setLog(log);
+            authConfigFactory = new AuthConfigFactory(log);
             imageConfigResolver.setLog(log);
 
             LogOutputSpecFactory logSpecFactory = new LogOutputSpecFactory(useColor, logStdout, logDate);

--- a/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/PushMojo.java
+++ b/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/PushMojo.java
@@ -61,11 +61,6 @@ public class PushMojo extends AbstractDockerMojo {
     }
 
     @Override
-    public void contextualize(Context context) throws ContextException {
-        authConfigFactory = new AuthConfigFactory();
-    }
-
-    @Override
     public void executeInternal(ServiceHub serviceHub) throws MojoExecutionException {
         if (skipPush) {
             return;

--- a/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/develop/WatchMojo.java
+++ b/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/develop/WatchMojo.java
@@ -161,11 +161,6 @@ public class WatchMojo extends AbstractDockerMojo {
     private ServiceHub hub;
 
     @Override
-    public void contextualize(Context context) throws ContextException {
-        authConfigFactory = new AuthConfigFactory();
-    }
-
-    @Override
     public void execute() throws MojoExecutionException, MojoFailureException {
         if (skip) {
             return;

--- a/openshift-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/AbstractDockerMojo.java
+++ b/openshift-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/AbstractDockerMojo.java
@@ -22,6 +22,7 @@ import org.codehaus.plexus.component.repository.exception.ComponentLookupExcepti
 import org.codehaus.plexus.context.Context;
 import org.codehaus.plexus.context.ContextException;
 import org.eclipse.jkube.generator.api.GeneratorContext;
+import org.eclipse.jkube.kit.build.api.auth.AuthConfig;
 import org.eclipse.jkube.kit.build.core.GavLabel;
 import org.eclipse.jkube.kit.build.core.JKubeBuildContext;
 import org.eclipse.jkube.kit.build.core.config.JKubeBuildConfiguration;
@@ -428,7 +429,6 @@ public abstract class AbstractDockerMojo extends AbstractMojo implements ConfigH
     @Override
     public void contextualize(Context context) throws ContextException {
         plexusContainer = ((PlexusContainer) context.get(PlexusConstants.PLEXUS_KEY));
-        authConfigFactory = new AuthConfigFactory();
     }
 
     @Override
@@ -436,13 +436,11 @@ public abstract class AbstractDockerMojo extends AbstractMojo implements ConfigH
         if (!skip) {
             boolean ansiRestore = Ansi.isEnabled();
             log = new AnsiLogger(getLog(), useColorForLogging(), verbose, !settings.getInteractiveMode(), getLogPrefix());
+            authConfigFactory = new AuthConfigFactory(log);
+            imageConfigResolver.setLog(log);
 
             try {
-                authConfigFactory.setLog(log);
-                imageConfigResolver.setLog(log);
-
                 LogOutputSpecFactory logSpecFactory = new LogOutputSpecFactory(useColor, logStdout, logDate);
-
 
                 DockerAccess access = null;
                 try {
@@ -574,7 +572,7 @@ public abstract class AbstractDockerMojo extends AbstractMojo implements ConfigH
         }
     }
 
-    protected RegistryService.RegistryConfig getRegistryConfig(String specificRegistry) throws MojoExecutionException {
+    protected RegistryService.RegistryConfig getRegistryConfig(String specificRegistry) {
         return new RegistryService.RegistryConfig.Builder()
                 .settings(MavenUtil.getRegistryServerFromMavenSettings(settings))
                 .authConfig(authConfig != null ? authConfig.toMap() : null)

--- a/openshift-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/BuildMojo.java
+++ b/openshift-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/BuildMojo.java
@@ -49,9 +49,6 @@ import static org.eclipse.jkube.kit.config.resource.RuntimeMode.kubernetes;
 @Mojo(name = "build", defaultPhase = LifecyclePhase.PRE_INTEGRATION_TEST, requiresDependencyResolution = ResolutionScope.COMPILE)
 public class BuildMojo extends AbstractDockerMojo implements Contextualizable {
 
-    // Handler dealing with authentication credentials
-    private AuthConfigFactory authConfigFactory;
-
     @Override
     protected boolean isDockerAccessRequired() {
         boolean ret = false;
@@ -72,11 +69,6 @@ public class BuildMojo extends AbstractDockerMojo implements Contextualizable {
         clusterAccess = new ClusterAccess(getClusterConfiguration());
         // Platform mode is already used in executeInternal()
         executeDockerBuild();
-    }
-
-    @Override
-    public void contextualize(Context context) throws ContextException {
-        authConfigFactory = new AuthConfigFactory();
     }
 
     @Override
@@ -114,7 +106,7 @@ public class BuildMojo extends AbstractDockerMojo implements Contextualizable {
     public void executeDockerBuild() throws MojoExecutionException, MojoFailureException {
         if (!skip) {
             log = new AnsiLogger(getLog(), useColor, verbose, !settings.getInteractiveMode(), getLogPrefix());
-            authConfigFactory.setLog(log);
+            authConfigFactory = new AuthConfigFactory(log);
             imageConfigResolver.setLog(log);
 
             LogOutputSpecFactory logSpecFactory = new LogOutputSpecFactory(useColor, logStdout, logDate);

--- a/openshift-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/develop/WatchMojo.java
+++ b/openshift-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/develop/WatchMojo.java
@@ -161,11 +161,6 @@ public class WatchMojo extends AbstractDockerMojo {
     private ServiceHub hub;
 
     @Override
-    public void contextualize(Context context) throws ContextException {
-        authConfigFactory = new AuthConfigFactory();
-    }
-
-    @Override
     public void execute() throws MojoExecutionException, MojoFailureException {
         if (skip) {
             return;

